### DISCUSSION TITLE
Publishing to WorkSpace and Scheduling Refreshes

### DISCUSSION THREAD

**Discussion by sycodon** [migueesc123/PowerBIRESTAPI#165] [General]
I was able to utilize this connector to great effect. But when I published it and tried to set up a refresh schedule I received an error:

_Unable to determine the data source due to unknown functions. When custom connectors are used, the error can happen if gateway 
doesn't have the extension enabled. Details: Static analysis failed in gateway. gatewayObjectId:f2b44c9c-b315-43f1-8e5e-e5c58c5d6342, resultCode:FailedWithUnknownFunctions Query contains unknown function name: PowerBIRESTAPI.Navigation Unknown function name: PowerBIRESTAPI.Navigation is a valid Mashup library function name format. Hence fail the operation. .
Data source for Query1_

I presume I need to contact the Gateway Admin (whoever that is) and request them to enable this extension. Is there anything in particular I need to tell them in addition to what is already detailed in the message.

I am also curious as to how this connector circumvents the need to he Bearer token.

**Comment by klinejordan** [comment]
Instructions for custom connectors with gateways are here: https://learn.microsoft.com/en-us/power-bi/connect-data/service-gateway-custom-connectors

**Reply by sycodon** [reply]
I don't' suppose there are videos on how to add this custom gateway? I have been given Fabric Admin access but have no idea how to proceed. The MS documentation is vague to say the least.

**Reply by klinejordan** [reply]
What is vague about it? Where are you stuck? What have you tried so far? 

Being a Fabric admin by itself isn't actually required to add a custom connector to a gateway. You need to be a Gateway Admin (but Fabric admins can make themselves a gateway admin via https://admin.powerplatform.microsoft.com/ext/DataGateways) to configure gateway settings, and you need to be able to log into the machine where the gateway software is enabled to add the .mez file to the folder where the gateway expects Custom Connector files.

**Reply by sycodon** [reply]
Well, that last sentence caught my attention. We have an existing Gateway and it has a populated path to a Custom connector folder, but the folder doesn't exist. I had not been sure of what should go there until just now. I'll make the folder and throw the file in and see what happens.

thx
